lib/v2/screens/receive_screens/receive_enter_data/receive_seeds_screen.dart

```diff
@@ -1,69 +1,107 @@
 import 'package:seeds/v2/components/amount_entry/amount_entry_widget.dart';
 import 'package:seeds/v2/components/balance_row.dart';
 import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/full_page_error_indicator.dart';
+import 'package:seeds/v2/components/full_page_loading_indicator.dart';
+import 'package:seeds/v2/components/snack_bar_info.dart';
 import 'package:seeds/v2/components/text_form_field_light.dart';
 import 'package:seeds/v2/domain-shared/ui_constants.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'interactor/receive_enter_data_bloc.dart';
+import 'interactor/viewmodels/receive_enter_data_events.dart';
+import 'interactor/viewmodels/receive_enter_data_state.dart';
 
 class ReceiveEnterDataScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        appBar: AppBar(title: const Text("Receive")),
-        body: Stack(
-          children: [
-            SingleChildScrollView(
-              child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: [
-                  const SizedBox(height: 100),
-                  AmountEntryWidget(
-                    onValueChange: (value) {
-                      // BlocProvider.of<ReceiveEnterDataPageBloc>(context)
-                      //     .add(OnAmountChange(amountChanged: value));
-                    },
-                    autoFocus: true,
-                  ),
-                  const SizedBox(height: 36),
-                  Padding(
-                    padding: const EdgeInsets.symmetric(horizontal: horizontalEdgePadding),
-                    child: Column(
+    return BlocProvider(
+      create: (context) =>
+          ReceiveEnterDataBloc(BlocProvider.of<RatesBloc>(context).state)..add(const LoadUserBalance()),
+      child: Scaffold(
+          appBar: AppBar(title: const Text("Receive")),
+          body: BlocConsumer<ReceiveEnterDataBloc, ReceiveEnterDataState>(
+              listenWhen: (_, current) => current.pageCommand != null,
+              listener: (context, state) {
+                if (state.pageCommand is NavigateToReceiveDetails) {
+                  //Navigate to Receive Detail Page
+                }
+                if (state.pageCommand is ShowTransactionFailSnackBar) {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBarInfo(title: 'Receive creation failed, try again.', context: context),
+                  );
+                }
+              },
+              builder: (context, ReceiveEnterDataState state) {
+                switch (state.pageState) {
+                  case PageState.initial:
+                    return const SizedBox.shrink();
+                  case PageState.loading:
+                    return const FullPageLoadingIndicator();
+                  case PageState.failure:
+                    return const FullPageErrorIndicator();
```

Comment:
hum, im not sure we want to show a full page error if the transaction fails? 
Thats why we show the snackbar and tell them to try again.

Reply:
I think the full page error could be if the device have no internet before loading the receive page?

Reply:
no internet will be handled app high level, not per screen.

Reply:
we dont want each screen to have to add implementation for when there is no internet

Reply:
full page failure was remove